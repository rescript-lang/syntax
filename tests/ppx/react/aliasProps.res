@@jsxConfig({version: 4, mode: "automatic"})

module C0 = {
  @react.component
  let make = (~priority as _, ~text="Test") => React.string(text)
}

module C1 = {
  @react.component
  let make = (~priority as p, ~text="Test") => React.string(p ++ text)
}

module C2 = {
  @react.component
  let make = (~foo as bar="") => React.string(bar)
}

module C3 = {
  @react.component
  let make = (~foo as bar="", ~a=bar, ~b) => {
    React.string(bar ++ a ++ b)
  }
}

module C4 = {
  @react.component
  let make = (~a as b, ~x=true) => <div> b </div>
}

module C5 = {
  @react.component
  let make = (~a as (x, y), ~z=3) => x + y + z
}

module C6 = {
  module type Comp = {
    @react.component
    let make: unit => React.element
  }

  @react.component
  let make = (~comp as module(Comp: Comp), ~x as (a, b)) => <Comp />
}
