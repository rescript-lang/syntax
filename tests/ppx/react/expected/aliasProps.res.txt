@@jsxConfig({version: 4, mode: "automatic"})

module C0 = {
  type props<'priority, 'text> = {priority: 'priority, text?: 'text}

  @react.component
  let make = (props: props<'priority, 'text>) => {
    let text = switch props.text {
    | Some(text) => text
    | None => "Test"
    }
    and _ = props.priority

    React.string(text)
  }
  let make = {
    let \"AliasProps$C0" = (props: props<_>) => make(props)

    \"AliasProps$C0"
  }
}

module C1 = {
  type props<'priority, 'text> = {priority: 'priority, text?: 'text}

  @react.component
  let make = (props: props<'priority, 'text>) => {
    let text = switch props.text {
    | Some(text) => text
    | None => "Test"
    }
    and p = props.priority

    React.string(p ++ text)
  }
  let make = {
    let \"AliasProps$C1" = (props: props<_>) => make(props)

    \"AliasProps$C1"
  }
}

module C2 = {
  type props<'foo> = {foo?: 'foo}

  @react.component
  let make = (props: props<'foo>) => {
    let bar = switch props.foo {
    | Some(foo) => foo
    | None => ""
    }

    ReactDOM.jsx("div", {})
  }
  let make = {
    let \"AliasProps$C2" = (props: props<_>) => make(props)

    \"AliasProps$C2"
  }
}
